numpy/_core/include/numpy/experimental_dtype_api.h

```diff
@@ -1,387 +0,0 @@
-    ((__get_default_descr *)(__experimental_dtype_api_table[6]))
-
-static inline PyArray_Descr *
-PyArray_GetDefaultDescr(PyArray_DTypeMeta *DType)
-{
-    if (DType->singleton != NULL) {
-        Py_INCREF(DType->singleton);
-        return DType->singleton;
-    }
-    return _PyArray_GetDefaultDescr(DType);
-}
-
-
-/*
- * NumPy's builtin DTypes:
- */
-#define PyArray_BoolDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[10])
-/* Integers */
-#define PyArray_ByteDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[11])
-#define PyArray_UByteDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[12])
-#define PyArray_ShortDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[13])
-#define PyArray_UShortDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[14])
-#define PyArray_IntDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[15])
-#define PyArray_UIntDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[16])
-#define PyArray_LongDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[17])
-#define PyArray_ULongDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[18])
-#define PyArray_LongLongDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[19])
-#define PyArray_ULongLongDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[20])
-/* Integer aliases */
-#define PyArray_Int8DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[21])
-#define PyArray_UInt8DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[22])
-#define PyArray_Int16DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[23])
-#define PyArray_UInt16DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[24])
-#define PyArray_Int32DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[25])
-#define PyArray_UInt32DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[26])
-#define PyArray_Int64DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[27])
-#define PyArray_UInt64DType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[28])
-#define PyArray_IntpDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[29])
-#define PyArray_UIntpDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[30])
-/* Floats */
-#define PyArray_HalfDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[31])
-#define PyArray_FloatDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[32])
-#define PyArray_DoubleDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[33])
-#define PyArray_LongDoubleDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[34])
-/* Complex */
-#define PyArray_CFloatDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[35])
-#define PyArray_CDoubleDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[36])
-#define PyArray_CLongDoubleDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[37])
-/* String/Bytes */
-#define PyArray_StringDType (*(PyArray_DTypeMeta *)__experimental_dtype_api_table[38])
```

Comment:
I never noticed until now but I think it's a bug that we exposed this name in the public DType API. I changed it to `PyArray_BytesDType` instead to avoid the python2 baggage, which is its internal name. This is being newly exposed so let's give it a name that makes sense in Python.